games/zombie.py

```diff
@@ -1,438 +1,335 @@
 
-    def kill_zombies(self, num_zombies, random_bullet_chance):
-        kill_zombie = False
-        for i in range(num_zombies):
-            if self.alive_zombies:
-                kill_zombie = True
-                shot_zombie_serial = random.choice(self.alive_zombies)
-                self.controller_opts[shot_zombie_serial][3] = 0
-                self.dead_zombies[shot_zombie_serial] = time.time() + self.get_kill_time()
-                self.alive_zombies.remove(shot_zombie_serial)
+        self.game_loop()
 
-        if kill_zombie:
-            self.reward(random_bullet_chance)
-
-    def reward(self, random_bullet_chance):
-        if (len(self.dead_zombies) + len(self.alive_zombies)) > 3:
-            random_bullet = random.choice(random_bullet_chance)
-            sound = False
-            for i in range(random_bullet):
-                sound = True
-                random_human = random.choice(self.humans)
-                if self.controller_opts[random_human][4] < 5:
-                    self.controller_opts[random_human][4] += 1
-            if sound:
-                self.pickup.start_effect()
-            #one in 5 chance of getting a weapon
-            #self.get_weapon(5)
+    '''
+    Override joust functions
+    '''
+    # @Override
+    # Add additional zombie sounds
+    def init_audio(self):
+        super().init_audio()
+
+        # TODO - should this all load ahead of time?
+        if self.play_audio:
+            self.pickup = Audio('audio/Zombie/sounds/pickup.wav')
+            self.ten_seconds = Audio('audio/Zombie/vox/' + self.voice + '/10 seconds left.wav')
+            self.thirty_seconds = Audio('audio/Zombie/vox/' + self.voice + '/30 seconds.wav')
+            self.one_minute = Audio('audio/Zombie/vox/' + self.voice + '/1 minute.wav')
+            self.three_minutes = Audio('audio/Zombie/vox/' + self.voice + '/3 minutes.wav')
+            # self.five_minutes = Audio('audio/Zombie/vox/' + self.voice + '/5 minutes.wav')
+            self.human_victory = Audio('audio/Zombie/vox/' + self.voice + '/human_victory.wav')
+            self.zombie_victory = Audio('audio/Zombie/vox/' + self.voice + '/zombie_victory.wav')
+            self.death = Audio('audio/Zombie/vox/' + self.voice + '/zombie_death.wav')
+            self.pistol = Audio('audio/Zombie/sounds/pistol.wav')
+            # self.shotgun = Audio('audio/Zombie/sounds/shotgun.wav')
+            self.molotov = Audio('audio/Zombie/sounds/molotov.wav')
+
+    '''
+    Override generic functions
+    '''
+    # @Override
+    # Set the initial opts
+    def init_moves(self):
+        super().init_moves()
+
+        # Setting up human team
+        for move_serial in self.moves:
+            self.humans.append(move_serial)
+
+    # @Override
+    # Killing initial humans
+    def before_game_loop(self):
+        super().before_game_loop()
+
+        # Kill first humans
+        for random_human in random.sample(set(self.humans), 2):
```

Comment:
getting the error here:
Traceback (most recent call last):
  File "/home/aaron/JoustMania/piparty.py", line 1215, in <module>
    piparty.game_loop()
  File "/home/aaron/JoustMania/piparty.py", line 697, in game_loop
    self.check_start_game()
  File "/home/aaron/JoustMania/piparty.py", line 960, in check_start_game
    self.start_game()
  File "/home/aaron/JoustMania/piparty.py", line 1125, in start_game
    zombie.Joust(moves=game_moves, command_queue=self.command_queue, ns=self.ns, red_on_kill=self.red_on_kill, \
  File "/home/aaron/JoustMania/games/zombie.py", line 61, in __init__
    self.game_loop()
  File "/home/aaron/JoustMania/games/game.py", line 388, in game_loop
    self.before_game_loop()
  File "/home/aaron/JoustMania/games/zombie.py", line 104, in before_game_loop
    for random_human in random.sample(set(self.humans), 2):
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/random.py", line 439, in sample
    raise TypeError("Population must be a sequence.  "
TypeError: Population must be a sequence.  For dicts or sets, use sorted(d).